### PR TITLE
Modify denominators to be specific for each clinical condition

### LINKED ISSUE #23: Add specific denominators for clinical pathways
Currently we only have accurate numerator values for the clinical conditions because we have not created specific denominators yet. To calculate accurate rates for each of the 7 conditions, we need to create 7 condition specific denominators that we add to [the measures](https://github.com/opensafely/pharmacy-first/blob/df003bd07b40b636905f297b32a392216b7c982a/analysis/measures_definition_pf_codes_conditions.py#L114-L145) for the conditions. 

For now, we should start with the first two criteria for defining the population (age and sex). Because we still need to select the correct codes for the defining the exclusion criteria we ignore this for now, this work is captured in #26.

| Condition                        | Age Range              | Sex   | ~~Exclusions~~                                                                 |
|----------------------------------|------------------------|-------|-----------------------------------------------------------------------------|
| Uncomplicated Urinary Tract Infection | 16 to 64 years         | Female | ~~Pregnant individuals, urinary catheter, recurrent UTI (2 episodes in last 6 months, or 3 episodes in last 12 months)~~ |
| Shingles                          | 18 years and over       | Any   | ~~Pregnant individuals~~                                                        |
| Impetigo                          | 1 year and over         | Any   | ~~Bullous impetigo, recurrent impetigo (2 or more episodes in the same year), pregnant individuals under 16 years~~     |
| Infected Insect Bites             | 1 year and over         | Any   | ~~Pregnant individuals under 16 years~~                                         |
| Acute Sore Throat                 | 5 years and over        | Any   | ~~Pregnant individuals under 16 years~~                                         |
| Acute Sinusitis                   | 12 years and over       | Any   | ~~Immunosuppressed individuals, chronic sinusitis (symptoms lasting more than 12 weeks), pregnant individuals under 16 years~~ |
| Acute Otitis Media                | 1 to 17 years           | Any   | ~~Recurrent acute otitis media (3 or more episodes in 6 months or four or more episodes in 12 months), pregnant individuals under 16 years~~ |

### Acceptance criteria

1. **Add the following denominators to `measures_definition_pf_codes_conditions.py`**: 
    - [x] `denominator_uncomplicated_uti`
    - [x] `denominator_shingles`
    - [x] `denominator_impetigo`
    - [x] `denominator_infected_insect_bites`
    - [x] `denominator_acute_sore_throat`
    - [x] `denominator_acute_sinusitis`
    - [x] `denominator_acute_otitis_media`
2. **Update the methods and results sections of the the report to include a very brief description of each denominator we implemented (and what's still missing). This could be the table above.**

### PR DESCRIPTION
**Pull request by viv3ckj**
Closes #23 
Denominators added (excluding exclusions). 
Will do commenting and tidying, with exclusions in next commit.

### CODE REVIEW

File: analysis/measures_definition_pf_codes_conditions.py [hunk 1]
```diff
@@ -145,6 +145,24 @@
...
 
+denominator_uncomplicated_uti = (age>=16) & (age<=64) & (patients.sex.is_in(["female"]))
+denominator_shingles = age>=18
+denominator_impetigo = age>=1
+denominator_infected_insect_bites = age>=1
+denominator_acute_sore_throat = age>=5
+denominator_acute_sinusitis = age>=12
+denominator_acute_otitis_media = (age>=1) & (age<=17)
+
+denominators = {
```
Comment (milanwiedemann) [inline analysis/measures_definition_pf_codes_conditions.py:156]: great implementation, I'd just make it a bit clearer that these are the denominators for the pharmacy first conditions

```suggestion
pf_condition_denominators = {
```

File: analysis/measures_definition_pf_codes_conditions.py [hunk 2]
```diff
@@ -173,7 +191,7 @@
         measures.define_measure(
             name=f"count_{condition_name}_by_{breakdown}",
             numerator=numerator,
-            denominator=denominator,
+            denominator=denominators[condition_name],
```
Comment (milanwiedemann) [inline analysis/measures_definition_pf_codes_conditions.py:194]: this looks like a great way to do it! but I think we also need to add this to the denominaotr of the overall measure?

```py
    measures.define_measure(
        name=f"count_{condition_name}",
        numerator=numerator,
        denominator=pf_condition_denominators[condition_name],
        intervals=months(monthly_intervals).starting_on(start_date),
    )
```

also updating the suggested name change here:

```suggestion
            denominator=pf_condition_denominators[condition_name],
```

File: analysis/measures_definition_pf_codes_conditions.py [hunk 3]
```diff
@@ -145,6 +145,24 @@
             intervals=months(monthly_intervals).starting_on(start_date),
         )
 
+denominator_uncomplicated_uti = (age>=16) & (age<=64) & (patients.sex.is_in(["female"]))
```
Comment (milanwiedemann) [inline analysis/measures_definition_pf_codes_conditions.py:148]: we should add a comment above this line explaining which part of the denominaotr rules we already implemented, and importantly also that we didnt to the exclusions yet.

**Review by milanwiedemann — APPROVED** [review]
this looks really great already 🚀 left some ideas for you to consider

**Comment by viv3ckj** [comment]
Thanks Milan, I've made changes based on your comments.

**Review by milanwiedemann — CHANGES_REQUESTED** [review]
I think we should implement the denominators (and tests like you mentioned in #32) first in a dataset definition before adding them here. I'll make PR with suggestions

**Review by milanwiedemann — APPROVED** [review]

**Pull request merged**